CMakeLists.txt

```diff
@@ -1,25 +1,68 @@
 cmake_minimum_required(VERSION 3.14)
-project(cradle)
+
+# CRADLE_IS_MAIN_PROJECT defines whether we are building CRADLE as the main
+# project or as a dependency of another project.
+if(NOT DEFINED CRADLE_IS_MAIN_PROJECT)
+    if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+        set(CRADLE_IS_MAIN_PROJECT ON)
+    else()
+        set(CRADLE_IS_MAIN_PROJECT OFF)
+    endif()
+endif()
+
+# Define an option for generating installation targets.
+option(CRADLE_INSTALL "Generate install targets" OFF)
+
+# When building as part of the installation of a released version, the version
+# number must be externally specified.
+set(CRADLE_VERSION "" CACHE STRING
```

Comment:
Is `CACHE` appropriate here?

Reply:
I believe it is. `CACHE` allows user settings (e.g., from the command line) to override what we're setting here.